src/tests/class_tests/openms_gui/source/GUI/TSGDialog_test.cpp

```diff
@@ -0,0 +1,367 @@
+  QVERIFY2(double(p.getValue("c_intensity")) == 1.23, "Parameter 'c_intensity' wasn't set correctly.");
+
+  QVERIFY2(string(p.getValue("add_x_ions")) == "true", "Parameter 'add_x_ions' wasn't set correctly.");
+  QVERIFY2(double(p.getValue("x_intensity")) == 1.23, "Parameter 'x_intensity' wasn't set correctly.");
+
+  QVERIFY2(string(p.getValue("add_y_ions")) == "true", "Parameter 'add_x_ions' wasn't set correctly.");
+  QVERIFY2(double(p.getValue("y_intensity")) == 1.23, "Parameter 'x_intensity' wasn't set correctly.");
+
+  QVERIFY2(string(p.getValue("add_z_ions")) == "true", "Parameter 'add_z_ions' wasn't set correctly.");
+  QVERIFY2(double(p.getValue("z_intensity")) == 1.23, "Parameter 'z_intensity' wasn't set correctly.");
+
+  QVERIFY2(string(p.getValue("add_losses")) == "true", "Parameter 'add_losses' wasn't set correctly.");
+  QVERIFY2(double(p.getValue("relative_loss_intensity")) == 0.16, "Parameter 'relative_loss_intensity' wasn't set correctly.");
+
+  QVERIFY2(string(p.getValue("add_abundant_immonium_ions")) == "true", "Parameter 'add_a_ions' wasn't set correctly.");
+  QVERIFY2(string(p.getValue("add_precursor_peaks")) == "true", "Parameter 'add_precursor_peaks' wasn't set correctly.");
+
+  // try the other isotope models
+  UI->model_none->click();
+  p.clear();
+  p = dialog_.getParam_();
+  QVERIFY2(string(p.getValue("isotope_model")) == "none", "Parameter 'isotope_model' wasn't set correctly. Expected 'none'.");
+
+  UI->model_fine->click();
+  UI->max_iso_prob_spinbox->setValue(16);
+  p.clear();
+  p = dialog_.getParam_();
+  QVERIFY2(string(p.getValue("isotope_model")) == "fine", "Parameter 'isotope_model' wasn't set correctly. Expected 'fine'.");
+  QVERIFY2(double(p.getValue("max_isotope_probability")) == 0.16, "Parameter 'max_isotope_probability' wasn't set correctly.");
+
+  // check remaining ions for RNA input
+  UI->seq_type->setCurrentText("RNA");
+  p.clear();
+  p = dialog_.getParam_();
+  QVERIFY2(string(p.getValue("add_a-B_ions")) == "true", "Parameter 'add_a-B_ions' wasn't set correctly.");
+  QVERIFY2(double(p.getValue("a-B_intensity")) == 1.23, "Parameter 'a-B_intensity' wasn't set correctly.");
+
+  QVERIFY2(string(p.getValue("add_d_ions")) == "true", "Parameter 'add_d_ions' wasn't set correctly.");
+  QVERIFY2(double(p.getValue("d_intensity")) == 1.23, "Parameter 'd_intensity' wasn't set correctly.");
+
+  QVERIFY2(string(p.getValue("add_w_ions")) == "true", "Parameter 'add_w_ions' wasn't set correctly.");
+  QVERIFY2(double(p.getValue("w_intensity")) == 1.23, "Parameter 'w_intensity' wasn't set correctly.");
+}
+
+void TestTSGDialog::testSpectrumCalculation()
+{
+  // Spectrum generation settings don't really matter, since the dialog test shouldn't test the generators.
+  // Hence, it is only checked if a spectrum was produced.
+  
+  UI->seq_type->setCurrentText("Peptide");
+  QTest::qWait(DELAY);
+  UI->seq_input->setText("PEPTIDE");
```

Comment:
can you annotate that this is the peptide sequence?!
I was confused at first...